lib/BitcastUtils.cpp

```diff
@@ -837,19 +839,53 @@ bool RemoveCstExprFromFunction(Function *F) {
     auto OperandId = WorkList.back().second;
     WorkList.pop_back();
 
+    IRBuilder<> B(I);
     auto InsertBefore = I;
     if (auto phi = dyn_cast<PHINode>(I)) {
       InsertBefore =
           phi->getIncomingBlock(phi->getIncomingValueNumForOperand(OperandId))
               ->getFirstNonPHI();
     }
 
-    auto Operand = dyn_cast<ConstantExpr>(I->getOperand(OperandId))
-                       ->getAsInstruction(InsertBefore);
-    CheckInstruction(Operand);
-    I->setOperand(OperandId, Operand);
+    if (auto CstStruct = dyn_cast<ConstantArray>(I->getOperand(OperandId))) {
```

Comment:
Not a great name choice.

```suggestion
    if (auto CstArray = dyn_cast<ConstantArray>(I->getOperand(OperandId))) {
```